base/multidimensional.jl

```diff
@@ -395,11 +422,8 @@ module IteratorsMD
     @inline to_indices(A, inds, I::Tuple{CartesianIndices{0},Vararg{Any}}) =
         (first(I), to_indices(A, inds, tail(I))...)
 
-    @inline function in(i::CartesianIndex{N}, r::CartesianIndices{N}) where {N}
-        _in(true, i.I, first(r).I, last(r).I)
-    end
-    _in(b, ::Tuple{}, ::Tuple{}, ::Tuple{}) = b
-    @inline _in(b, i, start, stop) = _in(b & (start[1] <= i[1] <= stop[1]), tail(i), tail(start), tail(stop))
+    @inline in(i::CartesianIndex, r::CartesianIndices) = false
+    @inline in(i::CartesianIndex{N}, r::CartesianIndices{N}) where {N} = all(map(in, i.I, r.indices))
```

Comment:
this gives a better performance:

```julia
julia> R = CartesianIndices((1:3, 1:4))
3×4 CartesianIndices{2, Tuple{UnitRange{Int64}, UnitRange{Int64}}}:
 CartesianIndex(1, 1)  CartesianIndex(1, 2)  …  CartesianIndex(1, 4)
 CartesianIndex(2, 1)  CartesianIndex(2, 2)     CartesianIndex(2, 4)
 CartesianIndex(3, 1)  CartesianIndex(3, 2)     CartesianIndex(3, 4)

julia> @btime $CartesianIndex(1) in $R
  0.046 ns (0 allocations: 0 bytes) # this PR
  1.429 ns (0 allocations: 0 bytes) # master

julia> @btime $CartesianIndex(1, 2) in $R
  0.046 ns (0 allocations: 0 bytes) # this PR/master
```

Reply:
The result is spurious; to take 1/20th of ns you'd need a 20GHz CPU (so that 1 clock cycle = 0.05ns), but AFAIK such CPUs don't exist. Try using the `setup=(i = CartesianIndex(1))` notation in BenchmarkTools and see if that helps.

Reply:
LOL it seems that is still the same:

```julia
julia> @btime i in $R setup=(i=CartesianIndex(1, 2))
  0.046 ns (0 allocations: 0 bytes)
true

julia> @btime i in $R setup=(i=CartesianIndex(1))
  0.044 ns (0 allocations: 0 bytes)
false
```

Should I revert the `in(i::CartesianIndex, r::CartesianIndices) = false` method?

Reply:
That one might be there to catch cases where someone asks whether a `CartesianIndex{3}` is an element in a `CartesianIndices{2}` or something.